src/components/SettlementButton.js

```diff
@@ -76,29 +76,67 @@ class SettlementButton extends React.Component {
                 icon: Expensicons.Wallet,
-                value: ReportUtils.isExpenseReport(this.props.iouReport) ? CONST.IOU.PAYMENT_TYPE.VBBA : CONST.IOU.PAYMENT_TYPE.EXPENSIFY,
+                value: CONST.IOU.PAYMENT_TYPE.EXPENSIFY,
+            },
+            [CONST.IOU.PAYMENT_TYPE.VBBA]: {
+                text: this.props.translate('iou.settleExpensify'),
+                icon: Expensicons.Wallet,
+                value: CONST.IOU.PAYMENT_TYPE.VBBA,
             },
             [CONST.IOU.PAYMENT_TYPE.PAYPAL_ME]: {
                 text: this.props.translate('iou.settlePaypalMe'),
                 icon: Expensicons.PayPal,
                 value: CONST.IOU.PAYMENT_TYPE.PAYPAL_ME,
             },
             [CONST.IOU.PAYMENT_TYPE.ELSEWHERE]: {
-                text: this.props.translate('iou.settleElsewhere'),
+                text: isExpenseReport ? this.props.translate('iou.payExpenseElsewhere') : this.props.translate('iou.settleElsewhere'),
                 icon: Expensicons.Cash,
                 value: CONST.IOU.PAYMENT_TYPE.ELSEWHERE,
             },
         };
-        if (!this.props.shouldShowPaymentOptions && this.props.nvp_lastPaymentMethod[this.props.policyID]) {
-            return [paymentMethods[this.props.nvp_lastPaymentMethod[this.props.policyID]]];
+        const canUseWallet =
+            !isExpenseReport && this.props.currency === CONST.CURRENCY.USD && Permissions.canUsePayWithExpensify(this.props.betas) && Permissions.canUseWallet(this.props.betas);
+
+        // To achieve the one tap pay experience we need to choose the correct payment type as default,
+        // if user already paid for some request or expense, let's use the last payment method or use default.
+        if (!this.props.shouldShowPaymentOptions) {
+            let paymentMethod = this.props.nvp_lastPaymentMethod[this.props.policyID];
+            if (!paymentMethod) {
+                // In case the user hasn't paid a request yet, let's default to VBBA payment type in case of expense reports
+                if (isExpenseReport) {
+                    paymentMethod = CONST.IOU.PAYMENT_TYPE.VBBA;
+                } else if (canUseWallet) {
+                    // If they have Wallet set up, use that payment method as default
+                    paymentMethod = CONST.IOU.PAYMENT_TYPE.EXPENSIFY;
+                } else {
+                    paymentMethod = CONST.IOU.PAYMENT_TYPE.ELSEWHERE;
+                }
+            }
+
+            // In case the last payment method has been PayPal, but this request is made in currency unsupported by Paypal, default to Elsewhere
+            if (paymentMethod === CONST.IOU.PAYMENT_TYPE.PAYPAL_ME && !_.includes(CONST.PAYPAL_SUPPORTED_CURRENCIES, this.props.currency)) {
+                paymentMethod = CONST.IOU.PAYMENT_TYPE.ELSEWHERE;
+            }
+
+            // No options are shown only in case of the report preview settlement button where we want to use same short button label for all the options
```

Comment:
I'm not sure that I understand this comment 😅

Reply:
:D Updated